target/linux/ath79/image/generic.mk

```diff
@@ -42,6 +42,37 @@ define Build/add-elecom-factory-initramfs
   fi
 endef
 
+define Build/mkmylofw
+  $(eval device_id=$(word 1,$(1)))
+  $(eval revision=$(word 2,$(1)))
+
+  # TODO is this acceptable?
+  $(call Build/append-kernel)
+  $(call Build/pad-to,$(BLOCKSIZE))
+  $(call Build/append-rootfs)
+  $(call Build/pad-rootfs)
+
+  # TODO
+  # u-boot tries to boot 0x9f680000 first
+  # by setting the size parameter, the corresponding block is cleared
+  # if the firmware won't reach it
+  # a better solution could be to conditionally add an empty block (-b)
+  # but that fails if the firmware is long enough by itself
+  $(eval size=6684672)
+  # u-boot rejects images with sizes that aren't aligned to 0x0000
+  $(eval filesize=$(shell \
+		wc -c $@ | cut -f1 -d' ' |\
```

Comment:
Hmm, i might be blind - but i do not see making you use of the results from `wc -c $@ | cut -f1 -d' '` here. Or am i missing something?

Reply:
they are piped into python where they are "padded" until block boundary.
if it's bigger than `size`, it's used for the image generation.

Reply:
now i get it :nerd_face: